script/add-grammar

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env ruby
+
+require "optparse"
+
+ROOT = File.expand_path("../../", __FILE__)
+
+
+# Break a repository URL into its separate components
+def parse_url(input)
+  hosts = "github\.com|bitbucket\.org|gitlab\.com"
+  
+  # HTTPS/HTTP link pointing to recognised hosts
+  if input =~ /^(?:https?:\/\/)?(?:[^.@]+@)?(?:www\.)?(#{hosts})\/([^\/]+)\/([^\/]+)/i
+    { host: $1.downcase(), user: $2, repo: $3.sub(/\.git$/, "") }
+  # SSH
+  elsif input =~ /^git@(#{hosts}):([^\/]+)\/([^\/]+)\.git$/i
+    { host: $1.downcase(), user: $2, repo: $3 }
+  # provider:user/repo
+  elsif input =~ /^(github|bitbucket|gitlab):\/?([^\/]+)\/([^\/]+)\/?$/i
+    { host: $1.downcase(), user: $2, repo: $3 }
+  # user/repo - Common GitHub shorthand
+  elsif input =~ /^\/?([^\/]+)\/([^\/]+)\/?$/
+    { host: "github.com", user: $1, repo: $2 }
+  else
+    raise "Unsupported URL: #{input}"
+  end
+end
+
+# Isolate the vendor-name component of a submodule path
+def parse_submodule(name)
+  name =~ /^(?:.*(?:vendor\/)?grammars\/)?([^\/]+)/i
+  path = "vendor/grammars/#{$1}"
+  unless File.exist?("#{ROOT}/" + path)
+    warn "Submodule '#{path}' does not exist. Aborting."
+    exit 1
+  end
+  path
+end
+
+# Print debugging feedback to STDOUT if running with --verbose
+def log(msg)
+  puts msg if $verbose
+end
+
+
+usage = """Usage:
+  #{$0} [--replace grammar] url
```

Comment:
Yeah, it looks like the OptionParser does much less than in Python :/
Seems weird to have to write a usage message by hand...